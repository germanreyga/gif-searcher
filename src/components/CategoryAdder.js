import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdder = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((previousCategories) => [
        inputValue.trim(),
        ...previousCategories,
      ]);

      setInputValue("");
    } else {
      window.alert("You must add a valid name");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="e.j. Hunter X Hunter"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      />

      <button type="submit">Add new category</button>
    </form>
  );
};

CategoryAdder.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
