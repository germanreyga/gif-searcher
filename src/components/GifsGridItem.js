import React from "react";

export const GifsGridItem = ({ id, title, url }) => {
  return (
    <div className="gif-card animate__animated animate__fadeInLeft">
      <img src={url} alt={title} />
      <p>{title ? title : `No title for this GIF`}</p>
    </div>
  );
};
