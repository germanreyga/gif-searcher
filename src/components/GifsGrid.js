import React from "react";
import { GifsGridItem } from "./GifsGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifsGrid = ({ category }) => {
  const { data: imgs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="category-name animate__animated animate__fadeIn">
        {category}
      </h3>
      {loading && (
        <p className="animate__animated animate__flash">"Loading images..."</p>
      )}
      {
        <div className="card-grid">
          {imgs.map((img) => {
            return <GifsGridItem key={img.id} {...img} />;
          })}
        </div>
      }
    </>
  );
};

GifsGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
