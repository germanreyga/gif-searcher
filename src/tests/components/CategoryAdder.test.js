import { CategoryAdder } from "../../components/CategoryAdder";
import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

describe("testing the <CategoryAdder/> component", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<CategoryAdder setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<CategoryAdder setCategories={setCategories} />);
  });

  test("should display the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should change the text box", () => {
    const input = wrapper.find("input");
    const value = "Hello world";
    input.simulate("change", {
      target: {
        value: value,
      },
    });
    expect(wrapper.find("p").text().trim()).toBe(value);
  });

  test("should not post information on submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("should call setCategories and clean the textbox", () => {
    // 1. Simulate inputChange
    const input = wrapper.find("input");
    const value = "Hello world";
    input.simulate("change", {
      target: {
        value: value,
      },
    });
    // 2. Simulate submit
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    // 3. setCategories should have been called
    expect(setCategories).toHaveBeenCalledTimes(1);

    // 4. The input values should be empty ('')
    expect(input.prop("value")).toBe("");
  });
});
