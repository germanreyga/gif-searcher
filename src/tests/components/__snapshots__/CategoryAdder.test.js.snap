// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the <CategoryAdder/> component should display the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="e.j. Hunter X Hunter"
    type="text"
    value=""
  />
  <button
    className="submit-category"
    type="submit"
  >
    Add new category
  </button>
</form>
`;
